system/Database/Forge.php

```diff
@@ -796,6 +796,11 @@ public function dropColumn(string $table, $columnNames)
             return false;
         }
 
+        // If _alterTable() return execute result instead of sql string, return.
+        if ((bool) $sql === true) {
```

Comment:
This change makes `$sql` more incomprehensible.
Why don't you rewrite the code so that SQLite3 just overrides the `dropColumn()` method?
I think using overriding is more simpler and easier to read.

Reply:
I'm on it.

Reply:
@kenjis I want to ask if those unit tests also fail with the original SQLite3 solution on your side?

Reply:
If you commit only the updated test code first, check like this would be easier.

```console
kenji:~/work/codeigniter/official/CodeIgniter4 (develop *$=)$ git diff
diff --git a/tests/system/Database/Live/ForgeTest.php b/tests/system/Database/Live/ForgeTest.php
index 7fccd6b1d2..5772aae954 100644
--- a/tests/system/Database/Live/ForgeTest.php
+++ b/tests/system/Database/Live/ForgeTest.php
@@ -1317,7 +1317,8 @@ final class ForgeTest extends CIUnitTestCase
 
         $this->assertTrue($this->db->fieldExists('name', 'forge_test_two'));
 
-        $this->forge->dropColumn('forge_test_two', 'name');
+        $result = $this->forge->dropColumn('forge_test_two', 'name');
+        $this->assertTrue($result);
 
         $this->db->resetDataCache();
 
kenji:~/work/codeigniter/official/CodeIgniter4 (develop *$=)$ vendor/bin/phpunit tests/system/Database/Live/ForgeTest.php PHPUnit 10.5.20 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.2.18
Configuration: /Users/kenji/work/codeigniter/official/CodeIgniter4/phpunit.xml

.....SSS...S...........S.........F..............                                                            48 / 48 (100%)

Time: 00:04.753, Memory: 20.00 MB

There was 1 failure:

1) CodeIgniter\Database\Live\ForgeTest::testDropColumn
Failed asserting that false is true.

/Users/kenji/work/codeigniter/official/CodeIgniter4/tests/system/Database/Live/ForgeTest.php:1321

FAILURES!
Tests: 48, Assertions: 111, Failures: 1, Skipped: 5.
```

Reply:
But the unit tests fail occur not only at ForgeTest but also at MetadataTest, so I want to know that the test run result of the whole Live folder.

Reply:
All tests pass in the current `develop`.
https://github.com/codeigniter4/CodeIgniter4/actions/workflows/test-phpunit.yml?query=branch%3Adevelop

Push the fixed implementation and run GitHub Actions if you want to check the results in other than your local environment.
I'm not sure what you really want to know.

Reply:
```
------ -------------------------------------------------------------------------------------- 
  Line   system/Database/SQLite3/Forge.php                                                     
 ------ -------------------------------------------------------------------------------------- 
  129    Method CodeIgniter\Database\SQLite3\Forge::_alterTable() should                       
         return array<int, string>|string|null but returns bool.                               
  154    Method CodeIgniter\Database\SQLite3\Forge::_alterTable() should                       
         return array<int, string>|string|null but returns bool.                               
  171    Method CodeIgniter\Database\SQLite3\Forge::dropColumn() has parameter                 
         $columnNames with no value type specified in iterable type array.                     
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
 ------ -------------------------------------------------------------------------------------- 
```

@kenjis Is there any way to fix this PHPStan analysis error without modifying the base Forge's _alterTable function's defined comment?

Or should I also return the boolean result as a string type?

Reply:
You cannot change the signature of `_alterTable()`.